hw/ip/otbn/dv/otbnsim/sim/dmem.py

```diff
@@ -48,11 +48,13 @@ def __init__(self) -> None:
                                .format(dmem_size))
 
         # We represent the contents of DMEM as a list of 32-bit words (unlike
-        # the RTL, which uses a 256-bit array). An entry in self.data is None
-        # if the word has invalid integrity bits and we'll get an error if we
-        # try to read it. Otherwise, we store the integer value.
+        # the RTL, which uses a 256-bit array). An entry in self.data is a
+        # tuple of (value, validity). The value is an integer representing the
+        # 32-bit word, and the validity is a boolean indicating whether the
+        # word has valid integrity bits. If the validity is False, we'll get
+        # an error if we try to read it.
         num_words = dmem_size // 4
-        self.data: List[Optional[int]] = [None] * num_words
+        self.data: List[Tuple[int, bool]] = [(0, True)] * num_words
```

Comment:
Hmm: is that right? The code that was here before set the data to be uninitialised. Don't you want something like `(0, False)` to represent that?